java-checks/src/main/java/org/sonar/java/checks/MembersDifferOnlyByCapitalizationCheck.java

```diff
@@ -96,7 +96,32 @@ private static boolean isOverriding(Symbol symbol) {
 
   private static boolean isInvalidMember(Symbol currentMember, Symbol knownMember) {
     if (!isOverriding(currentMember)) {
-      return differentTypes(currentMember, knownMember) ? sameVisibilityNotPrivate(currentMember, knownMember) : !sameName(currentMember, knownMember);
+      return differentTypes(currentMember, knownMember) ? invalidMethodAndVariable(currentMember, knownMember) : !sameName(currentMember, knownMember);
+    }
+    return false;
+  }
+
+  private static boolean invalidMethodAndVariable(Symbol currentMember, Symbol knownMember) {
+    Symbol methodSymbol = currentMember.isMethodSymbol() ? currentMember : knownMember;
+    Symbol variableSymbol = methodSymbol == currentMember ? knownMember : currentMember;
+    return sameVisibilityNotPrivate(currentMember, knownMember)
```

Comment:
I would first start the method by 
```
if(!sameVisibilityNotPrivate(currentMember, knownMember)) {
  return false;
}
```
as this operation does not requires knowing which one is the method or the variable.